rm-datasource/src/main/java/io/seata/rm/datasource/undo/AbstractUndoLogManager.java

```diff
@@ -392,5 +413,22 @@ protected abstract void insertUndoLogWithNormal(String xid, long branchId, Strin
      * @return
      * @throws SQLException
      */
-    protected abstract byte[] getRollbackInfo(ResultSet rs) throws SQLException;
+    protected byte[] getRollbackInfo(ResultSet rs) throws SQLException  {
+        byte[] rollbackInfo = rs.getBytes(ClientTableColumnsName.UNDO_LOG_ROLLBACK_INFO);
+
+        String rollbackInfoContext = rs.getString(ClientTableColumnsName.UNDO_LOG_CONTEXT);
+        Map<String, String> context = CollectionUtils.decodeMap(rollbackInfoContext);
+        CompressorType compressorType = CompressorType.getByName(context.get(UndoLogConstants.COMPRESSOR_TYPE_KEY));
```

Comment:
old version upgrade new version. compressorType may be not exists in database. need to consider?

Reply:
use Map#getOrDefault to compatible history data